packages/core/src/LottieView/utils.ts

```diff
@@ -0,0 +1,35 @@
+import { Image } from 'react-native';
+
+function parsePossibleSources(source):
+  | {
+      sourceURL?: string;
+      sourceJson?: string;
+      sourceName?: string;
+      sourceDotLottieURI?: string;
+    }
+  | undefined {
+  if (typeof source === 'string') {
+    return { sourceName: source };
+  }
+
+  if (typeof source === 'object' && !(source as any).uri) {
+    return { sourceJson: JSON.stringify(source) };
+  }
+
+  if (typeof source === 'object' && (source as any).uri) {
+    // uri contains .lottie extension return sourceDotLottieURI
+    if ((source as any).uri.includes('.lottie')) {
+      return { sourceDotLottieURI: (source as any).uri };
+    }
+
+    return { sourceURL: (source as any).uri };
```

Comment:
There's many occurrences of `(source as any).uri` here, can we eithe type it properly or assign it to `const uri = (source as any).uri` only once and then use `uri`?